lib/utils/log/slog_handler.go

```diff
@@ -553,6 +556,32 @@ func NewSlogJSONHandler(w io.Writer, cfg SlogJSONHandlerConfig) *SlogJSONHandler
 	}
 }
 
+const (
+	traceID = "trace_id"
+	spanID  = "span_id"
+)
+
+func addTracingContextToRecord(ctx context.Context, r *slog.Record) {
+	span := oteltrace.SpanFromContext(ctx)
+	if span == nil || !span.IsRecording() {
```

Comment:
IIRC if you check `IsRecording` after the Span has ended, this will return false:

> After a Span is ended, it SHOULD become non-recording and IsRecording SHOULD always return false. The one known exception to this is streaming implementations of the API that do not keep local state and cannot change the value of IsRecording after ending the span.

https://opentelemetry.io/docs/specs/otel/trace/api/#isrecording

I can imagine a scenario where a log message is omitted by a goroutine after the parent function returns, and the span is ended (e.g an audit log event failing to be submitted). There's probably some argument here about whether or not you'd still want to attribute that to the trace.

In addition, how is IsRecording/TraceID/SpanID impacted by the configured tracing provider ? If no tracing provider is configured, is this always false? I imagine it would still be convenient to show trace ID/span ID even if the tracing provider is disabled.

Reply:
👍  It's a shame that noop doesn't generate span or trace IDs. Perhaps that's something that can be adjusted down the line.